src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs

```diff
@@ -1251,7 +1251,7 @@ internal SqlError ProcessSNIError(TdsParserStateObject stateObj)
             // !=null       | == 0     | replace text left of errorMessage
             //
 
-            Debug.Assert(!ADP.IsEmpty(errorMessage), "Empty error message received from SNI");
+            Debug.Assert(!ADP.IsEmpty(errorMessage) || sniError.sniError != 0, "Empty error message received from SNI");
```

Comment:
(Just nitpicking.. ) probably it's better to do a check based on sniError.errorMessage directly, to avoid confusion.